src/PlotFunctionModal.ts

```diff
@@ -0,0 +1,121 @@
+import { App, Modal, Setting } from 'obsidian'
+import { HeaderOptions } from './types'
+
+export default class PlotFunctionModal extends Modal {
+  title: HeaderOptions['title']
+  xLabel: HeaderOptions['xLabel']
+  yLabel: HeaderOptions['yLabel']
+  bounds: HeaderOptions['bounds'] = [-10, 10, -10, 10]
+  disableZoom: HeaderOptions['disableZoom'] = false
+  grid: HeaderOptions['grid'] = true
+
+  functions: string
+
+  onSubmit: (result: HeaderOptions & { functions: string }) => void
+
+  constructor(
+    app: App,
+    onSubmit: (result: HeaderOptions & { functions: string }) => void
+  ) {
+    super(app)
+    this.onSubmit = onSubmit
+  }
+
+  static parseToPlot(result: HeaderOptions & { functions: string }): string {
+    return `
+\`\`\`functionplot
+---
+title: ${result.title}
+disableZoom: ${result.disableZoom}
+bounds: [${result.bounds}]
+grid: ${result.grid}
+xLabel: ${result.xLabel}
+yLabel: ${result.yLabel}
```

Comment:
Should be `result.<field> ?? "\'\'"` for `title`, `xLabel` and `yLabel` respectively.

Reply:
All done in the new commit. `functions` field also had the same issue.